be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -114,8 +124,51 @@ Status SegmentIterator::_prepare_seek() {
     return Status::OK();
 }
 
+Status SegmentIterator::_init_row_ranges() {
+    if (_lower_rowid == _upper_rowid) {
+        // no data return;
+        _row_ranges = RowRanges();
+        return Status::OK();
+    }
+
+    if (_opts.conditions != nullptr) {
+        RowRanges zone_map_row_ranges;
+        RETURN_IF_ERROR(_get_row_ranges_from_zone_map(&zone_map_row_ranges));
+        RowRanges::ranges_intersection(_row_ranges, zone_map_row_ranges, &_row_ranges);
+        // TODO(hkp): get row ranges from bloom filter and secondary index
+    }
+
+    if (!_row_ranges.is_empty()) {
```

Comment:
I think we should put this judge to init() function

Reply:
ok